src/components/network-map-tab.js

```diff
@@ -209,43 +209,159 @@ export const NetworkMapTab = ({
         []
     );
 
+    const getEquipmentsNotFoundIds = useCallback(
+        (foundEquipmentPositions, allEquipments) => {
+            const foundEquipmentsIds = Array.from(
+                foundEquipmentPositions.keys()
+            );
+
+            const notFoundEquipmentsIds = allEquipments
+                .filter((s) => !foundEquipmentsIds.includes(s.id))
+                .map((s) => s.id);
+
+            return notFoundEquipmentsIds;
+        },
+        []
+    );
+
+    const getMissingEquipmentsPositions = useCallback(
+        (notFoundEquipmentsIds, fetchEquipmentCB) => {
+            if (notFoundEquipmentsIds.length === 0) {
+                return Promise.resolve([]);
+            }
+
+            return fetchEquipmentCB(
+                studyUuid,
+                currentNode?.id,
+                notFoundEquipmentsIds
+            );
+        },
+        [studyUuid, currentNode?.id]
+    );
+
     const loadMapGeoData = useCallback(() => {
-        console.info(`Loading geo data of study '${studyUuid}'...`);
-        const substationPositions = fetchSubstationPositions(
-            studyUuid,
-            currentNode?.id
-        );
-        const linePositions = lineFullPath
-            ? fetchLinePositions(studyUuid, currentNode?.id)
-            : [];
-        setWaitingLoadGeoData(true);
-
-        Promise.all([substationPositions, linePositions])
-            .then((values) => {
-                const newGeoData = new GeoData();
-                newGeoData.setSubstationPositions(values[0]);
-                newGeoData.setLinePositions(values[1]);
-                setGeoData(newGeoData);
-                setWaitingLoadGeoData(false);
-            })
-            .catch(function (error) {
-                console.error(error.message);
-                setWaitingLoadGeoData(false);
-                setErrorMessage(
-                    intlRef.current.formatMessage(
-                        { id: 'geoDataLoadingFail' },
-                        { studyUuid: studyUuid }
-                    )
+        if (studyUuid && currentNode) {
+            if (
+                geoData &&
+                (geoData.substationPositionsById.size > 0 ||
+                    geoData.linePositionsById.size > 0)
+            ) {
+                const notFoundSubstationIds = getEquipmentsNotFoundIds(
+                    geoData.substationPositionsById,
+                    mapEquipments.getSubstations()
                 );
-            });
-        dispatch(resetMapReloaded());
+
+                const notFoundLinesIds = lineFullPath
+                    ? getEquipmentsNotFoundIds(
+                          geoData.linePositionsById,
+                          mapEquipments.getLines()
+                      )
+                    : [];
```

Comment:
If lineFullPath is disabled there can't be missing line geo data ?

Reply:
there's no line positions download when lineFullPath is off